lib/Test2/Formatter/TAP.pm

```diff
@@ -154,12 +154,35 @@ sub event_ok {
     $out .= "not " unless $e->{pass};
     $out .= "ok";
     $out .= " $num" if defined($num);
+
+    # The regex form is ~250ms, the index form is ~50ms
+    my @extra;
+    defined($name) && (
```

Comment:
I know this is ugly as hell, and hard to read, but the performance of doing it this way vs blindly applying the regexes is unbelievable. Using && and , and similar to make this all 1 expression is also significantly faster. Normally I would not advocate for this kind of micro-optimization, but this gets done for every single OK event rendered to TAP, even tiny fractions of a second add up over a large test suite.